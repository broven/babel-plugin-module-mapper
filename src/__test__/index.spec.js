//TODO: a as function
const babel = require('@babel/core');
const plugin = require('../index');

const opt = {
  babelrc: false,
  plugins: [
    [plugin, {
      origin: {
      a: ['destModule', 'a']
    }}]
  ]
};

var example = `
import {a, b} from 'origin';
`;
describe('import', () => {
  it('simple Usage', () => {
    const {code} = babel.transformSync(example, opt);
    expect(code).toMatchSnapshot();
  });
  it('with local name', () => {
    const {code} = babel.transformSync(`
    import {a as c, b} from 'origin';
    `, opt);
    expect(code).toMatchSnapshot();
  })
  it('rewrite to default', () => {
    const {code} = babel.transformSync(`
    import {a as c, b} from 'origin';
    `, {
      babelrc: false,
      plugins: [
        [plugin, {
          origin: {
          a: ['destModule', 'default']
        }}]
      ]
    });
    expect(code).toMatchSnapshot();
  })
  it ('only import one', () => {
    const {code} = babel.transformSync(`
    import { a } from 'origin';
    `, {
      babelrc: false,
      plugins: [
        [plugin, {
          origin: {
          a: ['destModule', 'default']
        }}]
      ]
    });
    expect(code).toMatchSnapshot();
  })
});
describe('imported', () => {
  it('import', () => {
    const {code} = babel.transformSync(`
    import {a, b} from 'origin';
    console.log(a.b);
    `, opt);
    expect(code).toMatchSnapshot();
  });
   it ('use default import name', () => {
    const {code} = babel.transformSync(`
    import originDefault from 'origin';
    console.log(originDefault.a);
    `, {
      babelrc: false,
      plugins: [
        [plugin, {
          origin: {
           originDefault: ['destModule', 'a']
        }}]
      ]
    });
    expect(code).toMatchSnapshot();
  });
  // it ('use default import name', () => {
  //   const {code} = babel.transformSync(`
  //   import originDefault from 'origin';
  //   console.log(originDefault.a);
  //   const { a } = originDefault;
  //   const testVar = originDefault.a;
  //   `, opt);
  //   expect(code).toMatchSnapshot();
  // });
})

describe('real world case', () => {
  it ('propTypes', () => {
    const {code} = babel.transformSync(`
    import Nerv, { PropTypes, PureComponent } from 'nervjs';
    let contextTypes = {
      onClose: PropTypes.func
    };
    `, {
      babelrc: false,
      plugins: [
        [plugin, {
          nervjs: {
            PropTypes: ['prop-types', 'default']
        }}]
      ]
    });
    expect(code).toMatchSnapshot();
  });
  it ('multiple property mapper', () => {
    const {code} = babel.transformSync(`
    import Nerv, { PropTypes, PureComponent } from 'nervjs';
    let contextTypes = {
      onClose: PropTypes.func
    };
    `, {
      babelrc: false,
      plugins: [
        [plugin, {
          nervjs: {
            PropTypes: ['prop-types', 'default'],
            Nerv: ['react', 'default', 'React'],
            PureComponent: ['react', 'PureComponent']
        }}]
      ]
    });
    expect(code).toMatchSnapshot();
  });
  it ('should not throw error', () => {
    const {code} = babel.transformSync(`
    import 'weui'
    import Nerv from 'nervjs'
    import PickerGroup from './picker-group'
    import classNames from 'classnames'
    import { TOP, LINE_HEIGHT } from './constant'
    import * as dateHandle from './date'
    import './style/index.scss'

    `, {
      babelrc: false,
      plugins: [
        [plugin, {
          nervjs: {
            PropTypes: ['prop-types', 'default'],
            Nerv: ['react', 'default', 'React'],
            PureComponent: ['react', 'PureComponent']
        }}]
      ]
    });
    expect(code).toMatchSnapshot();
  });
})
