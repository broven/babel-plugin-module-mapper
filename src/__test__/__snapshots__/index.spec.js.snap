// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import only import one 1`] = `"import a from \\"destModule\\";"`;

exports[`import rewrite to default 1`] = `
"import { b } from 'origin';
import c from \\"destModule\\";"
`;

exports[`import simple Usage 1`] = `
"import { b } from 'origin';
import { a } from \\"destModule\\";"
`;

exports[`import with local name 1`] = `
"import { b } from 'origin';
import { a as c } from \\"destModule\\";"
`;

exports[`imported import 1`] = `
"import { b } from 'origin';
import { a } from \\"destModule\\";
console.log(a.b);"
`;

exports[`imported use default import name 1`] = `
"import { a as originDefault } from \\"destModule\\";
console.log(originDefault.a);"
`;

exports[`real world case multiple property mapper 1`] = `
"import { PureComponent } from \\"react\\";
import PropTypes from \\"prop-types\\";
import React from \\"react\\";
let contextTypes = {
  onClose: PropTypes.func
};"
`;

exports[`real world case propTypes 1`] = `
"import Nerv, { PureComponent } from 'nervjs';
import PropTypes from \\"prop-types\\";
let contextTypes = {
  onClose: PropTypes.func
};"
`;

exports[`real world case should not throw error 1`] = `
"import 'weui';
import React from \\"react\\";
import PickerGroup from './picker-group';
import classNames from 'classnames';
import { TOP, LINE_HEIGHT } from './constant';
import * as dateHandle from './date';
import './style/index.scss';"
`;
